lib/internal/url.js

```diff
@@ -166,11 +267,11 @@ class URLSearchParams {
       // Do nothing
     } else if (typeof init === 'object' || typeof init === 'function') {
       const method = init[SymbolIterator];
-      if (method === this[SymbolIterator]) {
+      if (method === this[SymbolIterator] && #searchParams in init) {
```

Comment:
I think this is a performance killer. Is there any impact? And I didn't understand why we need this line.